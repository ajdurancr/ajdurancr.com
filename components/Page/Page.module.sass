@import '../../styles/_variables'

.container
  margin: 0
  width: 100%
  padding-top: $menuHeightMobile
