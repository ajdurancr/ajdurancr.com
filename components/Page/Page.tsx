import React from 'react'

import type { CfSection, CfSEO, CfSite } from '../../types/contentful'
import { Section } from '../Section'
import styles from './Page.module.sass'

export type PageProps = {
  appTheme?: string
  site: CfSite['fields']
  pageUrl: string
  name: string
  seo?: CfSEO['fields']
  sections?: CfSection[]
}

const Page: React.FC<PageProps> = ({ sections }) => (
  <div className={styles.container}>
    {sections?.map(({ sys, fields }) => (
      <Section key={sys.id} {...fields} />
    ))}
  </div>
)

export { Page }
