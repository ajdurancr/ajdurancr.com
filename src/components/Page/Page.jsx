import React from 'react'
import propTypes from 'prop-types'

import { Section } from '..'
import styles from './Page.styles.sass'

const Page = ({ sections }) => (
  <div className={styles.container}>
    {sections && sections.map(({ sys, fields }) => (
      <Section key={sys.id} {...fields} />
    ))}
  </div>
)

Page.defaultProps = {
  site: {},
  sections: [],
}

Page.propTypes = {
  sections: propTypes.arrayOf(propTypes.object),
  site: propTypes.shape({
    id: propTypes.number,
    name: propTypes.string,
  }),
}

export default Page
