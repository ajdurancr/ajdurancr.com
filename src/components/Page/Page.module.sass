@import '../../styles/_variables'
// @import '../../styles/_ajTheme'

.container
  margin: 0
  width: 100%
  padding-top: $menuHeightMobile
