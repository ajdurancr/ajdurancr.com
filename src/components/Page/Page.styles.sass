.title
  margin: 0
  width: 100%
  padding-top: 80px
  line-height: 1.15
  font-size: 48px
  text-align: center
