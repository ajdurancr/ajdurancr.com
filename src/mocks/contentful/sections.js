import components from './components'

export default [
  {
    fields: {
      components,
      name: 'Main Section',
    },
  },
]
