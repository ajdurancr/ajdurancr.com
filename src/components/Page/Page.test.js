import React from 'react'
import { shallow } from 'enzyme'

import Page from './Page'
import sections from '../../mocks/contentful/sections'

describe('Page', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Page name="Test Page" sections={sections} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders correctly with default props', () => {
    const wrapper = shallow(<Page name="Test Page" />)

    expect(wrapper).toMatchSnapshot()
  })
})
