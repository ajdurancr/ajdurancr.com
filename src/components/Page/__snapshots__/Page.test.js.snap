// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders correctly 1`] = `
<div
  className="container"
>
  <Section
    components={
      Array [
        Object {
          "fields": Object {
            "componentDescription": "This is a test component",
            "contentType": "Test Component",
            "name": "TestComponent",
            "props": Array [
              Object {
                "fields": Object {
                  "name": "title",
                  "value": "This is a test title",
                },
                "sys": Object {
                  "contentType": Object {
                    "sys": Object {
                      "id": "propText",
                    },
                  },
                  "id": "74gEMnRKSOxejniOwjkHYN",
                },
              },
              Object {
                "fields": Object {
                  "name": "children",
                  "value": Array [
                    Object {
                      "fields": Object {
                        "componentDescription": "This is a child component",
                        "contentType": "Test Component",
                        "name": "ChildComponent",
                      },
                      "sys": Object {
                        "contentType": Object {
                          "sys": Object {
                            "id": "component",
                          },
                        },
                        "id": "4gfjlQfm3x9ShZKO8gdYqx",
                      },
                    },
                  ],
                },
                "sys": Object {
                  "contentType": Object {
                    "sys": Object {
                      "id": "propComponents",
                    },
                  },
                  "id": "3hwiQ7OcvmctEdQPye6exu",
                },
              },
              Object {
                "fields": Object {
                  "name": "markdownField",
                  "value": "Test value",
                },
                "sys": Object {
                  "contentType": Object {
                    "sys": Object {
                      "id": "propMarkdown",
                    },
                  },
                  "id": "1RezpPBXf2buGYIccqQw0i",
                },
              },
            ],
          },
          "sys": Object {
            "contentType": Object {
              "sys": Object {
                "id": "component",
              },
            },
            "id": "6neuJJsr2mbdKli0VjWnCn",
          },
        },
      ]
    }
    key="sectionId"
    name="Main Section"
  />
</div>
`;

exports[`Page renders correctly with default props 1`] = `
<div
  className="container"
/>
`;
